elastalert/elastalert.py

```diff
@@ -624,6 +625,33 @@ def get_hits_aggregation(self, rule, starttime, endtime, index, query_key, term_
 
         self.thread_data.num_hits += res['hits']['total']
         return {endtime: payload}
+    
+    def get_adv_query_aggregation(self, rule, starttime, endtime, index, query_key, term_size=None):
+        rule_filter = copy.copy(rule['filter'])
+        base_query = self.get_query(
+            rule_filter,
+            starttime,
+            endtime,
+            timestamp_field=rule['timestamp_field'],
+            sort=False,
+            to_ts_func=rule['dt_to_ts'],
+        )
+        request = get_msearch_query(base_query,rule)
+        print("yoyo")
```

Comment:
How is this different from other hits aggregation method call? why separate `get_adv_query_aggregation` ?

Reply:
other aggregation query will build aggs query based on the params (which aggregation to be built)and query_key fields. We dont require to build aggs query here, direct query string query itself will work. SO it will be similar to get_hits methods (called for Any type of rule). But with aggregations in output.